### PR TITLE
ADOP update to new CUDA and torch versions

### PR DESCRIPTION
**Pull request by lfranke**
compiling instructions and changes for windows, take a look at src/REAMDE_WINDOWS.md for details

### CODE REVIEW

File: src/apps/adop_train.cpp [hunk 1]
```diff
@@ -570,24 +586,24 @@ class NeuralTrainer
                 for (int i = 0; i < result.image_ids.size(); ++i)
                 {
                     // In average only write 10 images
-                    if (Random::sampleBool(std::min(1.0, 10.0 / loader_size)))
+                   // if (Random::sampleBool(std::min(1.0, 10.0 / loader_size)))
```
Comment (darglein) [inline src/apps/adop_train.cpp:589]: Please revert, otherwise TB will gets thousands of images

File: src/lib/config.h [hunk 2]
```diff
@@ -18,11 +18,30 @@
...
+#        define _NP_EXT_API extern
+#    endif
+
+#else
+// UNIX, ignore _API
+#    define _NP_API
+#    define _NP_EXT_API
+#    define _NP_INLINE_API inline
+#endif
+
```
Comment (darglein) [inline src/lib/config.h:38]: this is strange here

File: src/lib/config.h [hunk 3]
```diff
@@ -18,11 +18,30 @@
...
+
 namespace Saiga
 {
 class Camera;
 }
-inline Saiga::Camera* camera;
+_NP_INLINE_API _NP_API Saiga::Camera* camera;
```
Comment (darglein) [inline src/lib/config.h:44]: can we just make the global variable not inline?

File: src/lib/data/NeuralScene.cpp [hunk 4]
```diff
@@ -283,15 +283,15 @@ void NeuralScene::LoadCheckpoint(const std::string& checkpoint_dir)
     {
         std::cout << "Load Checkpoint pose" << std::endl;
 
-        std::cout << "First pose before " << poses->Download().front() << std::endl;
+      //  std::cout << "First pose before " << poses->Download().front() << std::endl;
```
Comment (darglein) [inline src/lib/data/NeuralScene.cpp:286]: remove whole line

File: src/lib/data/NeuralScene.cpp [hunk 5]
```diff
@@ -283,15 +283,15 @@ void NeuralScene::LoadCheckpoint(const std::string& checkpoint_dir)
...
 
         torch::load(poses, checkpoint_prefix + "poses.pth");
 
         SAIGA_ASSERT(poses->poses_se3.size(0) == scene->frames.size());
         SAIGA_ASSERT(poses->poses_se3.dtype() == torch::kDouble);
         SAIGA_ASSERT(poses->tangent_poses.dtype() == torch::kDouble);
 
-        std::cout << "First pose after " << poses->Download().front() << std::endl;
+        //std::cout << "First pose after " << poses->Download().front() << std::endl;
```
Comment (darglein) [inline src/lib/data/NeuralScene.cpp:294]: remove full line

File: src/lib/rendering/PointRenderer.cu [hunk 6]
```diff
@@ -1648,11 +1648,18 @@ __global__ void ApplyTangent(Vec6* tangent, Sophus::SE3d* pose, int n)
...
     if (tid >= n) return;
 
+    //TODO!! 
     Vec6 t = tangent[tid];
     auto p = pose[tid];
-    p      = Sophus::se3_expd(t) * p;
+#ifdef _WIN32
+    Sophus::SE3d p2(Sophus::se3_expd(t) * p);
+    for (int i = 0; i < 7; ++i)
+        pose[tid].data()[i] = p2.data()[i];
```
Comment (darglein) [inline src/lib/rendering/PointRenderer.cu:1657]: is this really needed?

File: src/apps/adop_train.cpp (hunk 1 above)
Comment (lfranke) [reply src/apps/adop_train.cpp:589]: I will move the check to just catch TB. The goal was to write all test images every eval epoch to disc, not to clutter TB

File: src/lib/config.h (hunk 3 above)
Comment (lfranke) [reply src/lib/config.h:44]: Didn't work for me. Windows was a bit tricky with this part

**Pull request merged**